routing/turns_tts_text.cpp

```diff
@@ -62,11 +67,70 @@ string GetTtsText::GetTurnNotification(Notification const & notification) const
+
+    // We have to use wstring in order to operate on Unicode text
+    // First, let's get rid of unpronounceable symbols.
+    std::wstring streetOut = std::wstring_convert<std::codecvt_utf8<wchar_t>>().from_bytes(notification.m_nextStreet);
+    // Semicolons are between destinations
+    // and pronounced more like commas.
+    std::replace( streetOut.begin(), streetOut.end(), ';', ',');
+    // Open brackets have no pronunciation analogue
+    std::replace( streetOut.begin(), streetOut.end(), '[', ' ');
+    // Closed brackets end a highway and introduce the rest
+    std::replace( streetOut.begin(), streetOut.end(), ']', ':');
+    // An angle bracket is currently used to represent "to" a place
+    // Ideally we'd add a translation for "to" or create full format
+    // strings for better i18n support, like "turn onto %s toward %s"
+    std::replace( streetOut.begin(), streetOut.end(), '>', ',');
+
+    // Replace any full-stop characters to make TTS flow better.
+    // Full stops are: . (Period) or 。 (East Asian) or । (Hindi)
+    std::wregex fullStops(L"[\\.\\。\\।]");
+    std::wstring distOut = std::wstring_convert<std::codecvt_utf8<wchar_t>>().from_bytes(distStr);
+    distOut = std::regex_replace(distOut, fullStops, L"");
+
+    // If the turn direction with the key +_street exists for this locale, use it (like make_a_right_turn_street)
+    string const dirStreetStr = GetTextById(GetDirectionTextId(notification)+"_street");
+    LOG(LDEBUG, ("TTS GOT dirstreetstr:", dirStreetStr));
+    std::wstring dirOut;
+    if (!dirStreetStr.empty()) {
+      dirOut = std::wstring_convert<std::codecvt_utf8<wchar_t>>().from_bytes(dirStreetStr);
+    } else {
+      dirOut = std::wstring_convert<std::codecvt_utf8<wchar_t>>().from_bytes(dirStr);
+    }
+    dirOut = std::regex_replace(dirOut, fullStops, L"");
+
+    string const distDirOntoStreetStr = GetTextById("dist_direction_onto_street");
+    std::wstring distDirOntoStreetOut = std::wstring_convert<std::codecvt_utf8<wchar_t>>().from_bytes(distDirOntoStreetStr);
+
+    LOG(LDEBUG, ("TTS GOT:",
+      std::wstring_convert<std::codecvt_utf8<wchar_t>>().to_bytes(distDirOntoStreetOut).c_str(),
+      std::wstring_convert<std::codecvt_utf8<wchar_t>>().to_bytes(distOut).c_str(),
+      std::wstring_convert<std::codecvt_utf8<wchar_t>>().to_bytes(dirOut).c_str(),
+      std::wstring_convert<std::codecvt_utf8<wchar_t>>().to_bytes(streetOut).c_str()
+    ));
+
+    char ttsOut[1024];
+    snprintf(ttsOut, 1023,
+      std::wstring_convert<std::codecvt_utf8<wchar_t>>().to_bytes(distDirOntoStreetOut).c_str(),
+      std::wstring_convert<std::codecvt_utf8<wchar_t>>().to_bytes(distOut).c_str(),
+      std::wstring_convert<std::codecvt_utf8<wchar_t>>().to_bytes(dirOut).c_str(),
+      std::wstring_convert<std::codecvt_utf8<wchar_t>>().to_bytes(streetOut).c_str()
+    );
+
+    LOG(LDEBUG, ("TTS:", ttsOut));
+
+    return ttsOut;
+  }
+
+  LOG(LDEBUG, ("TTS:", distStr + " " + dirStr));
```

Comment:
Those debug logs are for you only and should be removed in the final versions. You may mark them with TODO.